### ISSUE TITLE
snap last item with ItemDecoration

### ISSUE THREAD

**Issue by imunique-ZJ** [TakuSemba/MultiSnapRecyclerView#22]
It was  able to snap last item at v1.3.4, however after upgrading to v2.0.1 it will bounce back to the previous item rather than focus on the last item.

![2019-12-26 09-00-47 的螢幕擷圖](https://user-images.githubusercontent.com/4104660/71452298-9e389800-27be-11ea-988c-6f3d5784c7d1.png)

**Comment by TakuSemba** [comment]
Thank you for reporting. I will take a look.

**Comment by imunique-ZJ** [comment]
By the way, I use a custom LayoutManager and a custom ItemDecorator to achieve center focusing like the image above. After expand 1 more pixel for width in ItemDecoration, v2.0.1 will focus the last item as expected.
```kt
class CenteredItemDecoration: RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        // default offsets
        outRect.left = 0
        outRect.top = 0
        outRect.right = 0
        outRect.bottom = 0

        val itemWidth = view.resources.displayMetrics.widthPixels / 5
        view.updateLayoutParams<ViewGroup.LayoutParams> { width = itemWidth }

        val additionalOffset = view.resources.displayMetrics.widthPixels / 2 - itemWidth / 2
        when (parent.getChildAdapterPosition(view)) {
            0 -> outRect.left = additionalOffset
            (parent.layoutManager?.itemCount ?: 0) - 1 -> outRect.right = additionalOffset // +1 here will make v2.0.1 focus  last item
        }
    }
}
```

**Comment by imunique-ZJ** [comment]
Hi, any updates here?

**Comment by TakuSemba** [comment]
Sorry for the late reply.
Unfortunately, I could not reproduce it on my own.
Can you reproduce it on [sample app](https://github.com/TakuSemba/MultiSnapRecyclerView/tree/master/app) in this repo?

**Comment by imunique-ZJ** [comment]
Hi, I tried the Horizontal RecyclerView of the sample app and change the first sample to 3 items and set `msrv_gravity="center"`, `msrv_interval="1"` and added the CenteredItemDecoration I provided above, it worked as expected. However, if I remove the **margin** of the list item, it will fail to focus the last item.

```
item_horizontal.xml
<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:app="http://schemas.android.com/apk/res-auto"
  android:orientation="horizontal"
  android:layout_width="wrap_content"
  android:layout_height="wrap_content"
  android:layout_marginLeft="8dp" <!---- remove margin ---->
  android:layout_marginRight="8dp" <!---- remove margin ---->
  app:cardUseCompatPadding="true"
  >
    ...
</androidx.cardview.widget.CardView >
```

![device-2020-01-16-142607](https://user-images.githubusercontent.com/4104660/72499312-6cba6600-386c-11ea-96ce-38d9b57c4296.png)

If I commented out the line
```
view.updateLayoutParams<ViewGroup.LayoutParams> { width = itemWidth }
```
 it will focus the last item but not at the center of the item.
![device-2020-01-16-142641](https://user-images.githubusercontent.com/4104660/72499321-70e68380-386c-11ea-91fb-4689cbdeb45c.png)

**Comment by TakuSemba** [comment]
Thank you for the information, I will try to look into it.

**Comment by TakuSemba** [comment]
I fixed it at the version of 2.0.2. 👍 Could you check if it does not happen?

**Comment by imunique-ZJ** [comment]
Hooray!! It works!!
